### PR TITLE
Update macos.sh

### PR DESCRIPTION
**Pull request by iigorkarpov**

### CODE REVIEW

File: packaging/installer/dependencies/macos.sh [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+}
+
+homebrew_install() {
+    # Save some resources if running under GHA
+    if [[ -v GITHUB_ACTIONS ]]; then {
+        brew install automake
+        brew install openssl@3
+    }
+    else {
+        brew install openssl@3
```
Comment (vlvkobal) [inline packaging/installer/dependencies/macos.sh:75]: Why can't we install `openssl` instead of `openssl@3`?

Comment (iigorkarpov) [reply packaging/installer/dependencies/macos.sh:75]: We can. Should we? OpenSSL v3 is now default for Homebrew, I just specify the version explicitly.

Comment (vlvkobal) [reply packaging/installer/dependencies/macos.sh:75]: > Should we?

No, if you specify the version intentionally.

Comment (Ferroin) [reply packaging/installer/dependencies/macos.sh:75]: I would vote for not requiring a version here. We build just fine against earlier versions, and we don’t want to lock the user to a particular version of OpenSSL.

**Review by Ferroin — CHANGES_REQUESTED** [review]
This all appears to work correctly AFAICT, but the shellcheck warnings need to be fixed before it could be merged, and it also needs rebased.

**Pull request closed without merging**